.github/workflows/build.yml

```diff
@@ -0,0 +1,89 @@
+name: Build wheels and deploy
+
+on:
+  workflow_dispatch:
+    inputs:
+      confirm_ref:
+        description: "Enter branch/tag name to deploy to PyPI, or leave blank to only build wheels."
```

Comment:
To me this reads as "I can type the name of the branch I want to deploy to PyPI here", but the intention seems to be that one should type the name of this branch to confirm. Maybe we can change the start to `Enter the name of this tag or branch to deploy ..."?

Reply:
Yeah, this could be better.  I struggled with this wording, especially because there's also a minor space constraint - if you look in the picture in the PR text above, you can see that this text gets wrapped into really short lines, so I didn't want to write something that would end up 5 lines long and have no-one read it.

Perhaps?
> Confirm branch name to deploy to PyPI, leave blank to only build wheels.

This is a wordsmithing problem, so I bet @ajgpitch will enjoy it too.

Reply:
`Confirm branch name to deploy to PyPI, leave blank to only build wheels.` looks good to me. Or perhaps `Enter branch name to confirm deploying to PyPI, leave blank to only build wheels`?

Maybe anything other than blank or the correct branch name should fail the build? That is very much a nice to have though.

Reply:
Yeah, it's probably better to hard fail if something's typed but it's the wrong thing.  Unfortunately, there's no way to do pre-processing on these values, so I'll have to change the "canary" job that just exists to provide a visible tick saying "will attempt to deploy" to explicitly kill the whole run if the string is non-empty and unequal to the branch name.

Reply:
Can a check on `github.repository` or `github.repository_owner` work as well? See related PR for Mitiq https://github.com/unitaryfund/mitiq/pull/597

Reply:
I think that making the trigger based on a label ("PyPI release") would be nice.

Reply:
The trigger is a manual click action - you have to press a button on GitHub so there's no danger of "accidentally" running it on a fork.

Reply:
Or were you suggesting we do it based on a label?  In this, the only way to get a build is to type in a particular branch name, which implies that you have to have produced a special branch/git tag for the release, so that might be quite similar to what you were asking?

Reply:
I quite like having to do a manual release rather than using a tag because it allows one to rerun the build if a problem occurs and to try things out without filling the github repository with tags named things like "test-457".

Reply:
^ This is the reason I went for this method in the end. I think when I originally proposed automatic deployment I talked about it being tag-based, but I think this is cleaner.

You can also force-push a tag that had a failed release attached to it, though - that's not doing any harm, because a tag isn't _truly_ public until the release is made successfully.

Reply:
> 
> 
> Can a check on `github.repository` or `github.repository_owner` work as well? See related PR for Mitiq [unitaryfund/mitiq#597](https://github.com/unitaryfund/mitiq/pull/597)

So this file uses a slightly different mechanism - in mitiq you're tying the release action to GitHub's "make release" button, whereas here I tie it to you manually running this action.  One major advantage to this method is that admins can build a full set of wheels _without_ making a public release, which is a bigger deal for us because our wheels involve very nontrivial compilation, so there's rather more chance of something going wrong.